### PR TITLE
Add search for application list

### PR DESCRIPTION
**Pull request by ozkanpakdil**
fix for https://github.com/codecentric/spring-boot-admin/issues/998 
if applications has more then 1 item. a textbox will appear on applications page.
when user enters a string applications will be filtered by it
PS: I wish there was a wiki page how to contribute. and small explanation how to debug. I did not know Vue was included and testing took mvn deploy every time. and no way to see js and html :) it was challenging.

### CODE REVIEW

File: spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue [hunk 1]
```diff
@@ -75,7 +86,12 @@
       statusGroups() {
         const byStatus = groupBy(this.applications, application => application.status);
         const list = transform(byStatus, (result, value, key) => {
-          result.push({status: key, applications: sortBy(value, [application => application.name])})
+          if(this.filter===''){
+            result.push({status: key, applications: sortBy(value, [application => application.name])})
+          }else{
+            const filteredResult=value.filter( val => val.name.includes(this.filter))
```
Comment (shershen08) [inline spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue]: I'd suggest adding conversion to same case letters:
`value.filter( val => val.name.toLowerCase().includes(this.filter.toLowerCase()))`

File: spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue [hunk 2]
```diff
@@ -31,6 +31,13 @@
       </div>
       <template v-if="applicationsInitialized">
         <applications-stats :applications="applications" />
+            <input
+              class="input"
+              type="search"
+              placeholder="Application filter"
+              v-model="filter"
+              v-if="applications.length>1"
+            >
```
Comment (shershen08) [inline spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue]: maybe add button to clear the filter in a common container
```
<div v-if="applications.length>1">
  <input  class="input" type="search" placeholder="Application filter" v-model="filter">
  <button v-show="this.filter!==''" @click="this. filter = ''">Clear</button>
</div>
```

Comment (ozkanpakdil) [reply spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue]: at right side of textbox there is big X. so this is covered.
image

File: spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue (hunk 1 above)
Comment (ozkanpakdil) [reply spring-boot-admin-server-ui/src/main/frontend/views/applications/index.vue]: this I added.

**Comment by marbon87** [comment]
It would be great if the filter could be controlled by a uri parameter. That allows to create a bookmark including a filter.

**Comment by ozkanpakdil** [comment]
> It would be great if the filter could be controlled by a uri parameter. That allows to create a bookmark including a filter.

development cycle in vuejs is just terrible. any small change took me a deploy to test it properly in browser. anyway now you can bookmark url like this http://localhost:8080/#/applications?filterparam=zz that filterparam will used to filter applications onload.

**Comment by marbon87** [comment]
Thanks!

**Comment by joshiste** [comment]
> development cycle in vuejs is just terrible. any small change took me a deploy to test it properly in browser

just start the npm build in watch mode and use our sample with dev profile. See https://github.com/codecentric/spring-boot-admin/blob/master/spring-boot-admin-server-ui/README.md

**Comment by joshiste** [comment]
@ozkanpakdil Thanks for your contribution. 

Your work and the feedback from @marbon87 was taken into account for bb5a9c43383c8bc03e6c56632d3c34e2d1b72220. 
The commit has a bit more elaborated solution which also allows to filter by url, tags and version and  also drills down to instances matching the filter.

**Pull request closed without merging**